commit 454ac9d56acb643c48222c71f2b0e63926783b17
Author: Andreas Dilger <adilger@whamcloud.com>
Date:   Tue Feb 7 21:11:21 2012 -0500

    libext2fs: quiet spurious group checksum errors
    
    If we have to read the backup group descriptor checksums, the UNINIT
    flags are cleared to ensure that all of the inodes in the filesystem
    are scanned.  However, the code that reset the UNINIT flags did not
    reset the group checksum, and this produced many spurious error
    messages in e2fsck.
    
    Group descriptor 0 checksum is invalid.  FIXED.
    Group descriptor 1 checksum is invalid.  FIXED.
    :
    :
    
    Recompute checksums after modifying group descriptors to avoid these
    error messages.  Remove expected error messages in f_illitable_flexbg.
    
    Signed-off-by: Andreas Dilger <adilger@whamcloud.com>
    Signed-off-by: Theodore Ts'o <tytso@mit.edu>

Index: e2fsprogs-1.41.12/lib/ext2fs/openfs.c
===================================================================
--- e2fsprogs-1.41.12.orig/lib/ext2fs/openfs.c
+++ e2fsprogs-1.41.12/lib/ext2fs/openfs.c
@@ -347,7 +347,11 @@ errcode_t ext2fs_open2(const char *name,
 			gd->bg_flags &= ~EXT2_BG_BLOCK_UNINIT;
 			gd->bg_flags &= ~EXT2_BG_INODE_UNINIT;
 			gd->bg_itable_unused = 0;
+			/* The checksum will be reset later, but fix it here
+			 * anyway to avoid printing a lot of spurious errors. */
+			ext2fs_group_desc_csum_set(fs, i);
 		}
+
 		ext2fs_mark_super_dirty(fs);
 	}
 
Index: e2fsprogs-1.41.12/tests/f_illitable_flexbg/expect.1
===================================================================
--- e2fsprogs-1.41.12.orig/tests/f_illitable_flexbg/expect.1
+++ e2fsprogs-1.41.12/tests/f_illitable_flexbg/expect.1
@@ -1,17 +1,13 @@
 ../e2fsck/e2fsck: Group descriptors look bad... trying backup blocks...
-One or more block group descriptor checksums are invalid.  Fix? yes
-
-Group descriptor 0 checksum is invalid.  FIXED.
 Inode table for group 1 is not in group.  (block 0)
 WARNING: SEVERE DATA LOSS POSSIBLE.
 Relocate? yes
 
-Group descriptor 1 checksum is invalid.  FIXED.
-Group descriptor 2 checksum is invalid.  FIXED.
-Group descriptor 3 checksum is invalid.  FIXED.
 Pass 1: Checking inodes, blocks, and sizes
 Relocating group 1's inode table to 142...
 Restarting e2fsck from the beginning...
+One or more block group descriptor checksums are invalid.  Fix? yes
+
 Group descriptor 1 checksum is invalid.  FIXED.
 Pass 1: Checking inodes, blocks, and sizes
 Pass 2: Checking directory structure
